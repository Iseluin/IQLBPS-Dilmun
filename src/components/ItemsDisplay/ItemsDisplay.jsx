import React, { useEffect } from 'react';
import { useDispatch, useSelector } from '../../app/store';
import { getItemList } from '../../features/slices/item';
import ItemsCard from '../ItemCard/ItemsCard';

const ItemsDisplay = () => {
  const { itemList } = useSelector((state) => state.item);
  const dispatch = useDispatch();


  useEffect(() => {
    dispatch(getItemList());
  }, []);
  return (
    <div
      className="w-full grid lg:col-span-2 xl:col-span-4 gap-y-4 xl:grid-cols-3 lg:grid-cols-2 md:grid-cols-2 grid-cols-1 content-center items-center justify-items-center bg-background py-6"
      data-testid="itemsdisplay-div"
    >
      {itemList.map((item) => {
        return <ItemsCard key={item.id} item={item} />;
      })}
    </div>
  );
};

export default ItemsDisplay;
