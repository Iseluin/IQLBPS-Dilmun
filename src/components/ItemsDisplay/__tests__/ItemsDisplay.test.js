import { render, screen } from '@testing-library/react';
import ItemsDisplay from '../ItemsDisplay';

test('should render feedbackCard component', () => {
  render(<ItemsDisplay />);
  expect(screen.getByTestId('itemsdisplay-div')).toMatchSnapshot();
});
