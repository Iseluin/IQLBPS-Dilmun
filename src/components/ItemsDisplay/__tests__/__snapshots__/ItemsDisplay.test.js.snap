// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render feedbackCard component 1`] = `
<div
  class="grid lg:grid-cols-4 lg:col-span-3 md:grid-cols-3 sm:grid-cols-2 grid-cols-1 gap-6"
  data-testid="itemsdisplay-div"
/>
`;
